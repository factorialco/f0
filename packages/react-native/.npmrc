# Monorepo configuration for React Native package
shamefully-hoist=true
strict-peer-dependencies=false
