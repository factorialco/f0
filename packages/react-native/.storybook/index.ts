import AsyncStorage from "@react-native-async-storage/async-storage";
import { view } from "./storybook.requires";

const StorybookUIRoot = view.getStorybookUI({
  storage: {
    getItem: AsyncStorage.getItem,
    setItem: AsyncStorage.setItem,
  },
  // Enable tab mode instead of drawer - much better for Android
  enableWebsockets: true,
  // Start with the navigator closed
  isUIHidden: false,
  // Use bottom tabs instead of sidebar
  shouldDisableKeyboardAvoidingView: true,
  // Make sure gestures work
  shouldPersistSelection: true,
});

export default StorybookUIRoot;
