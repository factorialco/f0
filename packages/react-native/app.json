{
  "expo": {
    "name": "F0 React Native Storybook",
    "slug": "f0-react-native-storybook",
    "version": "1.0.0",
    "orientation": "portrait",
    "icon": "./playground/assets/icon.svg",
    "userInterfaceStyle": "light",
    "splash": {
      "image": "./playground/assets/splash.png",
      "resizeMode": "contain",
      "backgroundColor": "#ffffff"
    },
    "assetBundlePatterns": ["**/*"],
    "ios": {
      "supportsTablet": true
    },
    "android": {
      "adaptiveIcon": {
        "foregroundImage": "./playground/assets/adaptive-icon.png",
        "backgroundColor": "#ffffff"
      }
    },
    "web": {
      "favicon": "./playground/assets/favicon.png"
    },
    "plugins": ["expo-router"],
    "experiments": {
      "typedRoutes": true
    }
  }
}
