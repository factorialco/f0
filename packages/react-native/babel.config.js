module.exports = function (api) {
  api.cache(true);
  return {
    presets: [
      ["babel-preset-expo", { jsxImportSource: "nativewind" }],
      "nativewind/babel",
    ],
    plugins: [
      [
        "module-resolver",
        {
          root: ["./src", "./playground"],
          alias: {
            "@": "./src",
            lib: "./playground/lib",
            ui: "./playground/ui",
          },
        },
      ],
      "react-native-reanimated/plugin",
    ],
  };
};
