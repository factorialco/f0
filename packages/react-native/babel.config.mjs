export default {
  presets: ["babel-preset-expo", "@babel/preset-typescript"],
};
