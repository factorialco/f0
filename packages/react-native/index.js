// Import side effects first and services

// Initialize services

// Register app entry through Expo Router
import "expo-router/entry";

import { registerRootComponent } from "expo";
import { ExpoRoot } from "expo-router";
import StorybookUIRoot from "./.storybook";
import "./playground/global.css";

// Check for STORYBOOK_ENABLED environment variable
// This allows using:
// - yarn dev:storybook - to run with Storybook enabled
// - yarn dev:app - to run the regular app
// - yarn dev - to use the default setting below
const SHOW_STORYBOOK = process.env.STORYBOOK_ENABLED === "true" || true;

// https://docs.expo.dev/router/reference/troubleshooting/#expo_router_app_root-not-defined

// Must be exported or Fast Refresh won't update the context
export function App() {
  const ctx = require.context("./playground/app");
  return <ExpoRoot context={ctx} />;
}

// Register the app or Storybook based on the flag
registerRootComponent(SHOW_STORYBOOK ? StorybookUIRoot : App);
