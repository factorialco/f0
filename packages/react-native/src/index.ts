// Export components
export * from "./components/Activity/ActivityItem";
export * from "./components/Button";
export * from "./components/ExampleComponent";
export * from "./components/Icon";
export * from "./components/Avatars/exports";
export * from "./components/Counter";
export * from "./components/OnePreset";
export * from "./components/Tags/exports";
export * from "./components/OneChip";
export * from "./components/Navigation/PageHeader";
export * from "./components/Tags/export";

// Export icons
export * from "./icons";

// Export modules
export * from "./modules/inbox/InboxView";
