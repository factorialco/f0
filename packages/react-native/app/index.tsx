import * as React from "react";
import { View } from "react-native";
import Animated, {
  FadeInUp,
  FadeOutDown,
  LayoutAnimationConfig,
} from "react-native-reanimated";
import { Info } from "../lib/icons/Info";

import { Text } from "ui/text";
import { Tooltip, TooltipContent, TooltipTrigger } from "ui/tooltip";
import {
  Card,
  CardContent,
  CardDescription,
  CardFooter,
  CardHeader,
  CardTitle,
} from "ui/card";
import { Avatar, AvatarImage, AvatarFallback } from "ui/avatar";
import { Progress } from "ui/progress";
import { Button } from "src/components/Button";

const GITHUB_AVATAR_URI =
  "https://i.pinimg.com/originals/ef/a2/8d/efa28d18a04e7fa40ed49eeb0ab660db.jpg";

export default function Screen() {
  const [progress, setProgress] = React.useState(78);

  function updateProgressValue() {
    setProgress(Math.floor(Math.random() * 100));
  }

  return (
    <View className="bg-secondary/30 flex-1 items-center justify-center gap-5 p-6">
      <Card className="w-full max-w-sm rounded-2xl p-6">
        <CardHeader className="items-center">
          <Avatar alt="Rick Sanchez's Avatar" className="h-24 w-24">
            <AvatarImage source={{ uri: GITHUB_AVATAR_URI }} />
            <AvatarFallback>
              <Text>RS</Text>
            </AvatarFallback>
          </Avatar>
          <View className="p-3" />
          <CardTitle className="text-5xl pb-2 text-center">
            Rick Sanchez
          </CardTitle>
          <View className="flex-row">
            <CardDescription className="text-base font-semibold">
              Scientist
            </CardDescription>
            <Tooltip delayDuration={150}>
              <TooltipTrigger className="px-2 pb-0.5 active:opacity-50">
                <Info size={14} strokeWidth={2.5} color="#71717a" />
              </TooltipTrigger>
              <TooltipContent className="px-4 py-2 shadow">
                <Text className="native:text-lg">Freelance</Text>
              </TooltipContent>
            </Tooltip>
          </View>
        </CardHeader>
        <CardContent>
          <View className="flex-row justify-around gap-3">
            <View className="items-center">
              <Text className="text-muted-foreground text-sm">Hoooo</Text>
              <Text className="text-xl font-semibold">C-137</Text>
            </View>
            <View className="items-center">
              <Text className="text-muted-foreground text-sm">Age</Text>
              <Text className="text-xl font-semibold">70</Text>
            </View>
            <View className="items-center">
              <Text className="text-muted-foreground text-sm">Species</Text>
              <Text className="text-xl font-semibold">Human</Text>
            </View>
          </View>
        </CardContent>
        <CardFooter className="flex-col gap-3 pb-0">
          <View className="flex-row items-center overflow-hidden">
            <Text className="text-muted-foreground text-sm">Productivity:</Text>
            <LayoutAnimationConfig skipEntering>
              <Animated.View
                key={progress}
                entering={FadeInUp}
                exiting={FadeOutDown}
                className="w-11 items-center"
              >
                <Text className="font-bold text-sky-600 text-sm">
                  {progress}%
                </Text>
              </Animated.View>
            </LayoutAnimationConfig>
          </View>
          <Progress
            value={progress}
            className="h-2"
            indicatorClassName="bg-sky-600"
          />
          <View />
          <Button
            variant="outline"
            label="Update"
            className="shadow-foreground/5 shadow"
            onPress={updateProgressValue}
          />
        </CardFooter>
      </Card>
    </View>
  );
}
