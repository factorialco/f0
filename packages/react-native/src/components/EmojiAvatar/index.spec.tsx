import { render } from "@testing-library/react-native";
import React from "react";
import { EmojiAvatar } from "./";

describe("Button", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("Snapshot - with emoji", () => {
    const { toJSON } = render(<EmojiAvatar emoji="ðŸ‘‹" />);

    expect(toJSON()).toMatchSnapshot();
  });

  it("Snapshot - with text emoji", () => {
    const { toJSON } = render(<EmojiAvatar emoji="test" />);

    expect(toJSON()).toMatchSnapshot();
  });

  it("Snapshot - different sizes", () => {
    const sizes = ["sm", "md", "lg"] as const;

    sizes.forEach((size) => {
      const { toJSON } = render(<EmojiAvatar emoji="ðŸ‘‹" size={size} />);
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
