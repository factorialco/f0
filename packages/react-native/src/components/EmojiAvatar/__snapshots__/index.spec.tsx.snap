// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot - different sizes 1`] = `
<View
  className="flex aspect-square items-center justify-center border border-solid border-f1-border-secondary bg-f1-background dark:bg-f1-background-inverse-secondary w-6 h-6 rounded-sm"
>
  <Text
    className="text-xs"
  >
    ðŸ‘‹
  </Text>
</View>
`;

exports[`Button Snapshot - different sizes 2`] = `
<View
  className="flex aspect-square items-center justify-center border border-solid border-f1-border-secondary bg-f1-background dark:bg-f1-background-inverse-secondary w-9 h-9 rounded-md"
>
  <Text
    className="text-sm"
  >
    ðŸ‘‹
  </Text>
</View>
`;

exports[`Button Snapshot - different sizes 3`] = `
<View
  className="flex aspect-square items-center justify-center border border-solid border-f1-border-secondary bg-f1-background dark:bg-f1-background-inverse-secondary w-10 h-10 rounded-lg"
>
  <Text
    className="text-md"
  >
    ðŸ‘‹
  </Text>
</View>
`;

exports[`Button Snapshot - with emoji 1`] = `
<View
  className="flex aspect-square items-center justify-center border border-solid border-f1-border-secondary bg-f1-background dark:bg-f1-background-inverse-secondary w-9 h-9 rounded-md"
>
  <Text
    className="text-sm"
  >
    ðŸ‘‹
  </Text>
</View>
`;

exports[`Button Snapshot - with text emoji 1`] = `
<View
  className="flex aspect-square items-center justify-center border border-solid border-f1-border-secondary bg-f1-background dark:bg-f1-background-inverse-secondary w-9 h-9 rounded-md"
>
  <Text
    className="text-sm"
  >
    test
  </Text>
</View>
`;
